SPTDataLoader/SPTDataLoaderExponentialTimer.m

```diff
@@ -0,0 +1,158 @@
+
+@interface SPTDataLoaderExponentialTimer ()
+
+@property (nonatomic, assign, readwrite) NSTimeInterval timeInterval;
+@property (nonatomic, assign, readonly) NSTimeInterval maxTime;
+@property (nonatomic, assign, readonly) NSTimeInterval initialTime;
+
+@property (nonatomic, assign, readonly) double jitter;
+@property (nonatomic, assign, readonly) double growFactor;
+
+@end
+
+
+#pragma mark - SPTDataLoaderExponentialTimer Implementation
+
+@implementation SPTDataLoaderExponentialTimer
+
+#pragma mark Creating an Exponential Timer Object
+
++ (instancetype)exponentialTimerWithInitialTime:(NSTimeInterval)initialTime
+                                        maxTime:(NSTimeInterval)maxTime
+{
+    return [self exponentialTimerWithInitialTime:initialTime maxTime:maxTime jitter:SPTDataLoaderExponentialTimerDefaultJitter];
+}
+
++ (instancetype)exponentialTimerWithInitialTime:(NSTimeInterval)initialTime
+                                        maxTime:(NSTimeInterval)maxTime
+                                         jitter:(double)jitter
+{
+    return [[self alloc] initWithInitialTime:initialTime
+                                     maxTime:maxTime
+                                  growFactor:SPTDataLoaderExponentialTimerDefaultGrow
+                                      jitter:jitter];
+}
+
+- (instancetype)initWithInitialTime:(NSTimeInterval)initialTime
+                            maxTime:(NSTimeInterval)maxTime
+                         growFactor:(double)growFactor
+                             jitter:(double)jitter
+{
+    self = [super init];
+    if (!self) {
+        return nil;
+    }
+    
+    _initialTime = initialTime;
+    _timeInterval = initialTime;
+    _maxTime = maxTime;
+    _growFactor = growFactor;
+    _jitter = jitter;
+  
+    return self;
+}
+
+#pragma mark Accessing and Updating the Delay Value
+
+- (void)reset
+{
+    self.timeInterval = self.initialTime;
+}
+
+- (NSTimeInterval)calculateNext
+{
+    NSTimeInterval nextTime = self.timeInterval * self.growFactor;
+    
+    if (nextTime > self.maxTime) {
+        nextTime = self.maxTime;
+    }
+    
+    if (self.jitter < 0.0001) {
+        self.timeInterval = nextTime;
+    } else {
+        const double sigma = self.jitter * nextTime;
+        self.timeInterval = [self.class normalWithMu:nextTime sigma:sigma];
+    }
+    
+    if (self.timeInterval > self.maxTime) {
+        self.timeInterval = self.maxTime;
+    }
+    
+    return self.timeInterval;
+}
+
+- (NSTimeInterval)timeIntervalAndCalculateNext
+{
+    const NSTimeInterval timeInterval = self.timeInterval;
+    [self calculateNext];
+
+    return timeInterval;
+}
+
+#pragma mark Calculating Exponential Backoff
+
+#define EXPT_MODULO ((u_int32_t)RAND_MAX)
+#define EXPT_MODULO_F64 ((double)(EXPT_MODULO))
+NS_INLINE double SPTExptRandom()
+{
+    return arc4random_uniform(EXPT_MODULO + 1);
+}
+
++ (NSTimeInterval)normalWithMu:(double)mu sigma:(double)sigma
```

Comment:
I didn’t change `mu` as I don’t know what it stands for. Or if it’s actually a term used by the algorithm mentioned in the comment below.